Prepare fixes

csweaver opened the pull request:
fixed minor issues I encountered when running the prepare command

1. `cellxgene prepare <file>` does nothing because no file is saved. I added an error for this case
2. There are some hidden requirements that I added to the requirements.txt file -- louvain and python-igraph

freeman-lab commented:
cool @csweaver ! so the "error" you're flagging was actually intended behavior :) a common use case for me, especially when comparing preprocessing choices, was to save plots (with the `--plotting` option) but not bother saving a file

i'd be fine with either
1) change this error you've added into a warning
2) making the _default_ for output be writing a file with the same name as the original with `prepared` appended, and still let you specify `--no-output` to prevent file writing

in other words, i want to preserve the ability to run `prepare` and not write a file, but either of the above would seem to preserve that ability and address your concern (running and accidentally forgetting to specify an output)

i think my preference is (1)

csweaver commented:
@freeman-lab a warning is fine! I just was really confused when I ran it and I kept getting the same errors about cells on the frontend. I had to dig into the code to figure out I just wasn't writing a file at all.

freeman-lab commented:
totally get that! let's do a warning, perfect

csweaver commented on server/cli/prepare.py:
```diff
@@ -112,7 +117,12 @@ def run_neighbors(adata):
+        try:
+            sc.tl.louvain(adata)
+        except ModuleNotFoundError:
+            click.echo("\nWarning: louvain module is not installed, no clusters will be calculated. "
```
This interrupts the click progress bar so I included a new line so that the warning is on its own line and not after the progress bar

csweaver commented:
Added one more fix before we merge this puppy. --overwrite was not hooked up. I added the code to overwrite the original file as well as a warning if the overwrite flag was selected on loom data.

freeman-lab commented:
just to capture an offline chat with @csweaver  re: the overwrite changes here: the _intended_ behavior was for `--overwrite` to act as a flag that allows overwriting if you specify an output file with the same name as the original file, not to actually trigger overwriting. 

Rationale: I erred on the side of making overwriting quite difficult. You always need to specify an output  file if you want to save, and if it's going to overwrite, you need to explicitly specify that. My worry about the behavior proposed here is that it makes overwriting too easy.

Maybe it'd be better to rename the flag to `--allow-overwrite`?

Open to other opinions! Could also keep the current behavior for now and wait till we do a proper UX review.
